com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/CreateAppStep.java

```diff
@@ -83,9 +88,14 @@ protected StepPhase executeStep(ExecutionWrapper execution) throws FileStorageEx
             // Check if an application with this name already exists (as a result of a previous
             // execution):
             CloudApplication existingApp = client.getApplication(app.getName(), false);
+
             // If the application doesn't exist, create it:
             if (existingApp == null) {
-                client.createApplication(appName, staging, diskQuota, memory, uris, Collections.emptyList());
+                if (app.getDockerInfo() != null) {
+                    execution.getStepLogger().info(Messages.CREATING_APP_FROM_DOCKER_IMAGE, app.getName(), app.getDockerInfo().getImage());
+                }
+                client.createApplication(appName, staging, diskQuota, memory, uris, Collections.emptyList(), app.getDockerInfo());

```

Comment:
Have you handled the update scenario? What would happen if a user wants to change his docker image/username/password?